include/gpio_monitor.hpp

```diff
@@ -96,15 +101,17 @@ class GpioMonitor
      * @brief constructor
      *
      * @param[in] i_sysCfgJsonObj - System config JSON Object.
+     * @param[in] i_worker - pointer to the worker object.
      * @param[in] i_ioContext - pointer to IO context object.
      */
     GpioMonitor(const nlohmann::json i_sysCfgJsonObj,
+                const std::shared_ptr<Worker>& i_worker,
                 const std::shared_ptr<boost::asio::io_context>& i_ioContext) :
         m_sysCfgJsonObj(i_sysCfgJsonObj)
     {
         if (!m_sysCfgJsonObj.empty())
         {
-            initHandlerForGpio(i_ioContext);
+            initHandlerForGpio(i_ioContext, i_worker);
```

Comment:
I am interested in knowing, what happens if the config JSON is empty? :)

Reply:
Does nothing. It just returns back

Reply:
Then you have object in existence that does nothing. Why would you have that?

Reply:
Updated
throws error in case of empty JSON.